IPython/lib/display.py

```diff
@@ -310,11 +319,14 @@ class YouTubeVideo(IFrame):
     will be inserted in the document.
     """
 
-    def __init__(self, id, width=400, height=300, **kwargs):
+    def __init__(self, id, width=400, height=300, allow_autoplay=False, **kwargs):
         self.id=id
         src = "https://www.youtube.com/embed/{0}".format(id)
+        if allow_autoplay:
+            extras = list(kwargs.get("extras", [])) + ['allow="autoplay"']
+            kwargs.update(autoplay=1, extras=extras)
```

Comment:
Why `autoplay=1`?

Reply:
`autoplay` is a parameter required to get an iframe that can autoplay. ([IFrame Player API](https://developers.google.com/youtube/player_parameters#Parameters))
Is it a bit undesirable that IPython source code includes parameters defined by external services?

As shown below, my purpose is satisfied as long as there is an argument `extras`, so this part can be removed.

```python
YouTubeVideo("id", autoplay=1, extras=['allow="autoplay"'])
```

I think argument `extras`, which allows to add attributes to iframe tags from outside, is in demand.

Reply:
No, I meant to ask: why specifically `kwargs.update(autoplay=1)` here, it looks like an unused kwarg? `IFrame` class, which is the base class for `YouTubeVideo` doesn't use any `autoplay` kwarg, I thought `extras=['allow="autoplay"']` is all that was needed here.

Reply:
`kwargs` will finally be passed to `self.params` in the `IFrame` class and used as a query parameter for API. So, `kwargs` is a useful argument.

https://github.com/ipython/ipython/blob/deeb968055d1a2b1590bdc132ce3e3ea420da929/IPython/lib/display.py#L259-L268

Reply:
Oh, I missed that, sorry. Nevermind then.